semgrep-core/src/naming/Naming_AST.ml

```diff
@@ -402,6 +403,42 @@ let params_of_parameters env xs =
            Some (H.str_of_ident id, resolved)
        | _ -> None)
 
+(* In Angular JS, we have some "Injectable" classes, which are marked with an
+   @Injectable decorator.
+   https://angular.io/guide/dependency-injection-in-action
+   These classes may reference parameters to the constructor of the class, outside
+   of the actual code of the constructor itself.
+   So we must add them to the scope, should we find the decorator and a constructor's
+   parameters.
+   This also works for `@Component`.
+*)
+let js_add_angular_constructor_args env attrs defs =
+  let ( let* ) = Option.bind in
+  match
+    let* () =
+      List.find_map
+        (function
+          | NamedAttr (_, Id ((("Injectable" | "Component"), _), _), _) ->
+              Some ()
+          | _ -> None)
+        attrs
+    in
+    List.find_map
```

Comment:
If you used `List.filter_map` instead of `List.find_map` you wouldn't need the `match ... with`.